lib/core/alerts.ts

```diff
@@ -1,9 +1,54 @@
 import chalk from "chalk";
 
-const error = (message: string) => console.log(chalk.red(message));
-const warn = (message: string) => console.log(chalk.yellowBright(message));
-const notice = (message: string) => console.log(chalk.gray(message));
-const info = (message: string) => console.log(chalk.blueBright(message));
-const success = (message: string) => console.log(chalk.green(message));
+export type LogLevel = "verbose" | "error" | "info" | "silent";
+export const LOG_LEVELS: LogLevel[] = ["verbose", "error", "info", "silent"];
+
+export const logLevelDefault: LogLevel = "verbose";
+
+let currentLogLevel: LogLevel | undefined;
+
+export const setAlertsLogLevel = (logLevel: LogLevel) => {
+  currentLogLevel = logLevel;
+};
+
+type CbFunc = (...args: any) => any;
+type WrappedCbFunc<T extends CbFunc> = (
+  ...args: Parameters<T>
+) => ReturnType<T> | void;
+/**
+ * wraps a callback and only calls it if currentLogLevel is undefined or included in permittedLogLevels
+ * @param permittedLogLevels list of log levels. callbacks will only be called if current log level is listed here
+ * @param cb callback
+ */
+const withLogLevelsRestriction = <T extends CbFunc>(
+  permittedLogLevels: LogLevel[],
+  cb: T
+): WrappedCbFunc<T> => (...args: Parameters<T>): ReturnType<T> | void => {
+  const shouldCall =
+    !currentLogLevel || permittedLogLevels.includes(currentLogLevel);
+
+  if (shouldCall) {
+    return cb(...args);
+  }
+};
+
+const error = withLogLevelsRestriction(
+  ["verbose", "error", "info"],
+  (message: string) => console.log(chalk.red(message))
+);
+const warn = withLogLevelsRestriction(["verbose"], (message: string) =>
+  console.log(chalk.yellowBright(message))
```

Comment:
Prettier really insists on this inconsistent formatting, although it looks really bad to me

Reply:
Sometimes, if you manually break things onto newlines it will respect that but I think that mostly applies to objects, not functions: https://prettier.io/docs/en/rationale.html#multi-line-objects

It may not be ideal, but I don't think it's bad. Much nicer to have a tool determine it for us! 😅